applications/FluidDynamicsApplication/custom_processes/apply_mach_dependent_boundary_conditions.h

```diff
@@ -0,0 +1,292 @@
+   */
+  class KRATOS_API(FLUID_DYNAMICS_APPLICATION) ApplyMachDependentBoundaryConditions: public Process
+  {
+  public:
+        ///@name Type Definitions
+        ///@{
+        //
+        typedef ModelPart::NodeType NodeType;
+        
+        /**
+         * @brief This class validates and manages a variable to fix
+         * and the value to fix it to.
+         * 
+         * Method ActivateIfInsideTimeInterval must be called at the begining
+         * of the time step to see if the BC is active or not. This will cause
+         * the following behaviour in the other two public methods:
+         * - Enforce:
+         *     - If t inside interval  -> Fix dof
+         *     - If t outside interval -> Free dof
+         * - Release
+         *     - Always -> Free dof
+         * 
+         * This allows checking the time only once at the beginning of the
+         * time-step rather than at every node.
+         */
+        class BoundaryConditionUtility
+        {
+        public:
+            BoundaryConditionUtility(const std::string & variable_name,
+                                     const double Value,
+                                     const IntervalUtility & rIntervalUtility);
+            /**
+             * @brief Checks that the current time is within the interval, and
+             * decides accordingly what function to bind to mEnforceInternal
+             * (either FreeDof or FixDof).
+             */
+            void ActivateIfInsideTimeInterval(const double time);
+
+            /**
+             * @brief Calls mEnforceInternal. This will either fix or free the
+             * Dof depending on the result of ActivateIfInsideTimeInterval.
+             */
+            void Enforce(NodeType & rNode) const;
+
+            /// @brief Frees the Dof.
+            void Release(NodeType & rNode) const;
+
+        private:
+            const Variable<double> * mpVariable;
+            const double mValue; // Value to enforce
+            IntervalUtility mInterval;
+            
+            /* @brief Fixes the Dof managed by this class and sets the value
+             * to the one specified by mValue
+             */
+            static void FixDof(const BoundaryConditionUtility & rUtility, NodeType & rNode);
+
+            /// @brief Frees the Dof managed by this class
+            static void FreeDof(const BoundaryConditionUtility & rUtility, NodeType & rNode);
+
+            /** @brief This object points to the proper (Fix|Free)Dof function
+             * according to the time interval.
+             */
+            std::function<decltype(FixDof)> mEnforceInternal = FreeDof;
+        };
+
+        /// Pointer definition of ApplyMachDependentBoundaryConditions
+        KRATOS_CLASS_POINTER_DEFINITION(ApplyMachDependentBoundaryConditions);
+
+        ///@}
+        ///@name Life Cycle
+        ///@{
+
+        /// Constructor
+        ApplyMachDependentBoundaryConditions(Model& rModel, Parameters& rParameters);
```

Comment:
You do not need to pass Parameters as a reference. They are kind of pointers already, so you can send them by copy,

Reply:
True. Fixed now.